lib/core/httpRouter/routePart.js

```diff
@@ -73,10 +72,25 @@ class RoutePart {
    * @return {RouteHandler} registered function handler
    */
   getHandler(message) {
-    const
-      parsed = URL.parse(message.url, true),
-      pathname = parsed.pathname || '', // pathname is set to null if empty
-      routeHandler = new RouteHandler(pathname, parsed.query, message);
+    // we need a fake URL base because the WHATWG API currently doesn't handle
+    // relative URLs, and the http.IncomingMessage class doesn't provide an
+    // easy way (and an inexpensive one) of getting an absolute URL
+    // See: https://github.com/nodejs/node/issues/12682
+    const parsed = new URL(
+      message.url,
+      message.url[0] === '/' ? 'http://fake' : null);
+
+    // limit calls to "parsed.pathname", it's a getter joining an array
+    let pathname = parsed.pathname;
+
+    if (pathname[pathname.length - 1] === '/') {
+      pathname = pathname.slice(0, -1);
+    }
+
+    const qs = {};
+    parsed.searchParams.forEach((v, k) => (qs[k] = v));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment of "qs[k]" from this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AAssignmentWithinCondition)